### PR TITLE
Issue #603 Solved :- Adding Earliest Deadline First Scheduling Algorithm.

### LINKED ISSUE #603: [Feature Request] Adding Scheduling Algorithm - Earliest Deadline First.
### Is your feature request related to a problem? Please describe.
when processes in real-time systems miss their deadlines due to improper scheduling. In many scenarios, it becomes challenging to manage tasks with varying burst times and deadlines, leading to inefficient CPU utilization and potential system failures.

### Describe the solution you'd like

I would like an implementation of the Earliest Deadline First (EDF) scheduling algorithm that dynamically prioritizes processes based on their deadlines. The algorithm should ensure that processes with the nearest deadlines are executed first, minimizing the chances of deadline misses and improving the overall responsiveness of the system.

### Describe alternatives you've considered
However, these alternatives can lead to suboptimal performance in dynamic environments where the process arrival times and deadlines vary frequently. Additionally, I've looked into Rate Monotonic Scheduling (RMS), but it has limitations regarding periodic tasks and may not be as effective in all real-time scenarios compared to EDF.

---

**Would you like to work on this feature?**  
- [✅ ] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by PrAyAg9**
### Description
I would like an implementation of the Earliest Deadline First (EDF) scheduling algorithm that dynamically prioritizes processes based on their deadlines. The algorithm should ensure that processes with the nearest deadlines are executed first, minimizing the chances of deadline misses and improving the overall responsiveness of the system.

Fixes #603 
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [✅ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [ ✅] My code follows the style guidelines of this project
- [✅ ] I have performed a self-review of my code
- [ ✅] I have commented my code, particularly in hard-to-understand areas
- [ ✅] I have added tests that prove my fix is effective or that my feature works
- [ ✅] New and existing unit tests pass locally with my changes
- [ ✅] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/Scheduling Algorithms/EarliestDeadlineFirst.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+id: earliest-deadline-first
```
Comment (ajay-dhangar) [inline docs/algorithms/Scheduling Algorithms/EarliestDeadlineFirst.md:3]: add before `id` ``` ---```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code

**Comment by PrAyAg9** [comment]
@ajay-dhangar Done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**